src/main/java/ru/mercuriev/game/of/life/graph/cell/CellBuilder.java

```diff
@@ -0,0 +1,179 @@
+package ru.mercuriev.game.of.life.graph.cell;
+
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * This stands for object which:
+ * 1. holds the pointer to the current cell ({@link Cell}) while constructing new Graph
+ * 2. appends new cells to the graph
+ * 3. merging together current and specified CellBuilder
+ * Each time any method of the object is called current cell is the right and bottom most cell in graph (*1*)
+ *
+ * Class is not thread safe - do not use it without proper synchronization (e.g. in parallel stream )
+ *
+ * @author paul
+ */
+final class CellBuilder {
+
+    /**
+     * current cell
+     */
+    private Cell cell = null;
+
+    public CellBuilder append(int value) {
+
+        if (cell == null) {
```

Comment:
Success way IMHO is to return `CellBuilder` type but not `Cell`, as you are suggesting. Look carefully, please, to the method spec. This class is named Cell**Builder** so all methods bilding cell return *this*

Considering this, replacing 
```
        if (cell == null) {
            cell = new Cell(value);
        } else {
            Cell nextCell = new Cell(value);
            cell.right = nextCell;
            nextCell.left = cell;
            cell = cell.right;
        }
        return this;
```
to
```
        if (cell == null) {
            cell = new Cell(value);
            return this;
        } 
        Cell nextCell = new Cell(value);
        cell.right = nextCell;
        nextCell.left = cell;
        cell = cell.right;
        return this;
```
does not seem to be a good refactoring to me